### DISCUSSION TITLE
Are there plans of implementing a client generator?

### DISCUSSION THREAD

**Discussion by mpldr** [99designs/gqlgen#1978] [Q&A] [answered]
### What happened?

I was wondering if it was possible to add a subroutine for generating a matching GQL Client. It would be great to have a "default" way of consuming the API without having to either start implementing from the ground up, or using other libraries where one has to constantly pick the lesser evil.

**Comment by dclipca** [comment]
Use ```graphql-codegen```. Is this not enough?

**Comment by dclipca** [comment]
```graphql-codegen``` works for by introspecting the Graphql data.

Use
```
h.Use(extension.Introspection{})
```

then proceed with ```graphql-codegen```. They have very good docs.

**Comment by davidspek** [comment, marked as answer]
As far as I can find `graphql-codegen` doesn't support creating a Go client. However, I did find https://github.com/Yamashou/gqlgenc which is based on gqlgen and generates a Go client. I haven't used it yet though so no idea if it is any good.
